contracts/contracts/RequestManager.sol

```diff
@@ -606,25 +630,25 @@ contract RequestManager is Ownable, LpWhitelist, RestrictedCalls, Pausable {
 
         if (claimValid) {
             // If claim is valid, all stakes go to the claimer
-            ethToTransfer = claimerStake + challengerStakeTotal;
             claimReceiver = claimer;
+            ethToTransfer = claimerStake + challengerStakeTotal;
         } else if (challengerStakeTotal > 0) {
-            // If claim is invalid, partial withdrawal by the sender
-            ethToTransfer = 2 * claim.challengersStakes[msg.sender];
-            claimReceiver = msg.sender;
+            // If claim is invalid, partial withdrawal by the claimResolver
+            claimReceiver = claimResolver;
+            ethToTransfer = 2 * claim.challengersStakes[claimReceiver];
```

Comment:
Why are we multiplying with 2 here and saying it is partial withdraw?

Reply:
> Why are we multiplying with 2 here and saying it is partial withdraw?

Perhaps I can answer this. We support multiple challengers, i.e. anyone can join the challenge game:
```
(time flows to the right)
claimer -> ch1 -> claimer -> ch2 -> claimer -> ch1 -> claimer -> ch3 ...
```
Here each `chN` denotes a challenger. Claimer and challengers take turns in the game, but when the challengers get their turn, anyone can actually join in and make a challenge. In this case, we see `ch2` joining and challenging as a second challenger.

At the end of the game, each challenger has to call `withdraw` in order to get their stakes back, _plus_ the corresponding amount from the claimer's total stake. This is where both `partial` and `2 *` come from.
Partial, because each challenger only withdraws for themselves, i.e. they can only withdraw their part of the total stake. `2 *` because, in addition to their own stake that they put in during the game (`claim.challengersStakes[claimReceiver]`), they also get the same amount from the claimer's total stake. This is because, in order to win the game, the claimer has to match or exceed every challenger's stake.

Hope this helps.

Reply:
thanks @istankovic